packages/ketcher-core/src/application/editor/actions/erase.ts

```diff
@@ -106,16 +102,6 @@ export function fromFragmentDeletion(restruct, selection) {
     texts: selection.texts || []
   }
 
-  const actionRemoveDataSGroups = new Action()
-  restruct.molecule.sgroups.forEach((sg, id) => {
-    if (
-      selection.sgroupData.includes(id) ||
-      new Pile(selection.atoms).isSuperset(new Pile(sg.atoms))
-    ) {
-      actionRemoveDataSGroups.mergeWith(fromSgroupDeletion(restruct, id))
-    }
-  })
-
   selection.atoms.forEach((aid) => {
```

Comment:
Removal of data s-group should be taken care of by "removeSgroupIfNeeded" function that is called later: https://github.com/epam/ketcher/pull/2283/files#diff-4830423edc7e30690751de9bd936f4f65337856de30e83de31a0adcf650ddce6L148